arwen/host/arwen.go

```diff
@@ -262,10 +263,19 @@ func (host *vmHost) InitState() {
 
 // Close will close all underlying processes
 func (host *vmHost) Close() error {
+	host.mutExecution.Lock()
+	host.instanceInClose = true
 	host.runtimeContext.ClearWarmInstanceCache()
+	host.mutExecution.Unlock()
 	return nil
 }
 
+// Reset is a function which closes the VM and resets the instanceInClose variable
+func (host *vmHost) Reset() {
+	_ = host.Close()
+	host.instanceInClose = false
```

Comment:
suggestion to change a little bit the code in the following way (in order to prevent slight chances of concurrency issues in tests or wherever we might use this reset function:
```GO
func (host *vmHost) close(){
	host.runtimeContext.ClearWarmInstanceCache()
}

// Close will close all underlying processes
func (host *vmHost) Close() error {
	host.mutExecution.Lock()
	host.close()
        host.closingInstance = true
        host.mutExecution.Unlock()

       return nil
}

// Reset is a function which closes the VM and resets the instanceInClose variable
func (host *vmHost) Reset() {
        host.mutExecution.Lock()
	host.close()
        // keep closingInstance flag to false
        host.mutExecution.Unlock()
}
```

Reply:
nice input. added.